src/OpenTelemetry.Contrib.Instrumentation.AWSLambda/Implementation/AWSLambdaUtils.cs

```diff
@@ -112,5 +179,31 @@ private static ActivityContext ParseXRayTraceHeader(string rawHeader)
             var propagationContext = xrayPropagator.Extract(default, carrier, Getter);
             return propagationContext.ActivityContext;
         }
+
+        private static IEnumerable<string> GetHeaderValues(APIGatewayProxyRequest request, string name)
+        {
+            if (request.MultiValueHeaders != null &&
+                request.MultiValueHeaders.TryGetValue(name, out var values))
+            {
+                return values;
+            }
+
+            return null;
+        }
+
+        private static IEnumerable<string> GetHeaderValues(APIGatewayHttpApiV2ProxyRequest request, string name)
+        {
+            if (request.Headers != null &&
+                request.Headers.TryGetValue(name, out var header))
+            {
+                // Multiple values for the same header will be separated by a comma.
+                if (!string.IsNullOrEmpty(header))
```

Comment:
```suggestion
                if (header != null)
```

And could then be simplified to `return header?.Split(',');` without `if`.

Reply:
Yes, if we accept empty header as above then we even can do just `return header?.Split(',');` without `if`.

Reply:
See https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/408#discussion_r932954117